tests/phpunit/CRM/MembershipExtras/Queue/Builder/OfflineMultipleInstalmentPlansTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+use CRM_MembershipExtras_Queue_TestCase as QueueTestCase;
+use CRM_MembershipExtras_Test_Entity_PaymentPlanMembershipOrder as PaymentPlanMembershipOrder;
+use CRM_MembershipExtras_Test_Fabricator_MembershipType as MembershipTypeFabricator;
+use CRM_MembershipExtras_Test_Fabricator_PaymentPlanOrder as PaymentPlanOrderFabricator;
+use CRM_MembershipExtras_Queue_Builder_OfflineMultipleInstalmentPlans as OfflineMultipleInstalmentPlansQueueBuilder;
+
+/**
+ * Class CRM_MembershipExtras_Queue_Builder_MultipleInstalmentPlansTest
+ *
+ * @group headless
+ */
+class CRM_MembershipExtras_Queue_Builder_MultipleInstalmentPlansTest extends QueueTestCase {
+
+  /**
+   * A rolling membership type that we
+   * will use for creating payment plan orders.
+   *
+   * @var array
+   */
+  private $testRollingMembershipType;
+
+  /**
+   * The PriceFieldValue entity for the
+   * test rolling membership type.
+   *
+   * @var array
+   */
+  private $testRollingMembershipTypePriceFieldValue;
+
+  public function setUp() {
+    parent::setUp();
+    $this->setDefaultPaymentPlanSettings();
+    $this->createTestRollingMembershipType();
+  }
+
+  private function setDefaultPaymentPlanSettings() {
+    civicrm_api3('Setting', 'create', [
+      'membershipextras_paymentplan_days_to_renew_in_advance' => 0,
+    ]);
+
+    civicrm_api3('Setting', 'create', [
+      'membershipextras_paymentplan_update_start_date_renewal' => 0,
+    ]);
+  }
+
+  private function createTestRollingMembershipType() {
+    $this->testRollingMembershipType = MembershipTypeFabricator::fabricate(
+      [
+        'name' => 'Test Rolling Membership',
+        'period_type' => 'rolling',
+        'minimum_fee' => 120,
+        'duration_interval' => 1,
+        'duration_unit' => 'year',
+      ]);
+
+    $this->testRollingMembershipTypePriceFieldValue = civicrm_api3('PriceFieldValue', 'get', [
+      'sequential' => 1,
+      'membership_type_id' => $this->testRollingMembershipType['id'],
+      'options' => ['limit' => 1],
+    ])['values'][0];
+  }
+
+  public function testBuildQueueWillAddTasksToQueue() {
+    for ($i = 0; $i < 20; $i++) {
```

Comment:
would be better to make this 12, it would save some time of creating more payment plans and the test should still pass.